src/corehost/corehost.cpp

```diff
@@ -0,0 +1,85 @@
+#include "common/trace.h"
+#include "common/pal.h"
+#include "common/utils.h"
+
+enum StatusCode
+{
+    Failure                  = 0x87FE0000,
```

Comment:
Some POSIX syscalls make it difficult to retrieve exit codes over 255 for processes, so we should try to limit our return codes to that range (see [`wait(2)`](http://man7.org/linux/man-pages/man2/wait.2.html), specifically `wait` and `waitpid`, and the WEXITSTATUS macro). Also, some codes have semi-predefined use (see [`sysexits(3)`](https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man3/sysexits.3.html)) though these are not very standardized. We should at least ensure that we avoid the standard-ish exit codes unless our use matches their definition.